hytra/util/progressbar.py

```diff
@@ -25,3 +25,60 @@ def show(self, increase=1):
                 sys.stdout.flush()
         except IOError:
             pass
+
+class DefaultProgressVisitor:
+    def __init__(self):
+        pass
+
+    def setState(self,name):
+        pass
+
+    def showState(self,name):
+        pass
+
+    def showProgress(self, pos):
+        pass
+
+class CommandLineProgressVisitor(DefaultProgressVisitor):
+    def __init__(self, start=0, stop=1.0):
+        self.state = "State"
+        self._state = start
+        self._start = start
+        self._stop = stop
+
+    def setState(self,name):
+        self.state=name
+
+    def showState(self,name="State"):
+        if not name=="State":
+            self.state=name
+        try:
+            sys.stdout.write(self.state+'\n')
+        except IOError:
+            pass
+
+    def showProgress(self,pos):#increase=1):
+        self._state = pos #increase
+        if self._state > self._stop:
+            self._state = self._stop
+
+        pos = float(pos)
+        try:
+            sys.stdout.write("\r[%-20s] %d%%" % ('='*int(20*pos), (100*pos)))
+
+            if self._state >= self._stop:
+                sys.stdout.write('\n')
+                sys.stdout.flush()
+        except IOError:
+            pass
+
+class ProgressVisitor(DefaultProgressVisitor):
+    def __init__(self):
+        pass
+
+    def setProgressVisitor(self,progressVisitor):
```

Comment:
I don't get the `ProgressVisitor`. It subclasses `DefaultProgressVisitor` which provides only empty methods, so setting `self.progressVisitor` here doesn't have any consequences, same as setting all the other members. I think that class should be removed. Also, it is not used from the ilastik side but there you did what one would expect -- you derived a special visitor for the Gui progress bar.